dhall/src/Dhall/TypeCheck.hs

```diff
@@ -195,7 +196,9 @@ typeWithA tpa = loop
             loop ctx b2
 
     loop ctx e@(Annot x t       ) = do
-        _ <- loop ctx t
+        case Dhall.Core.denote t of
+            Const Sort -> return ()
```

Comment:
Would that be a good idea?

```suggestion
            Const _ -> return ()
```

Reply:
Yeah, there's no reason to type-check `t` further if it is already a `Const`